### PR TITLE
PurgedKFold::split leaks: test window starts at the first test label's END time

### LINKED ISSUE #65: PurgedKFold::split leaks: test window starts at the first test label's END time
## Problem

`crates/openquant/src/cross_validation.rs`, `PurgedKFold::split` (lines ~150-151 on `main` @ d3e6dfa):

```rust
let test_start = self.samples_info_sets[test_indices[0]].1;
let test_end   = self.samples_info_sets[*test_indices.last().unwrap()].1;
```

`samples_info_sets` is `(label_start, label_end)`. `test_start` is read from `.1` — the **end** of the first test label — so the purge window begins too late. mlfinlab's `PurgedKFold` (AFML snippet 7.3) builds the window from the **start** time of the first test sample to the **max** end time across the fold.

## Consequence: label leakage at the leading edge of every fold

Daily bars, 3-day labels, test fold beginning at sample 3 with span `[t3, t6]`:

- window used today: `[t6, ...]`
- training sample 2 has span `[t2, t5]`: `start_in` false (t2 < t6), `end_in` false (t5 < t6), `envelop` false → **kept in training**, although its label overlaps test sample 3's label on t3..t5. Same for sample 1 (`[t1, t4]`).

With `pct_embargo > 0` the symmetric "before" embargo in this implementation can hide the leak when the embargo happens to be at least as long as the label horizon; with `pct_embargo = 0`, or labels longer than the embargo, it leaks.

Secondary: `test_end` uses the last sample's end rather than the max end over the fold. With variable-length (triple-barrier) labels an earlier test sample can end later than the last one, so the trailing edge can under-purge too.

## Why the tests didn't catch it

`tests/cross_validation.rs` uses inline data with no external reference (see #40).

## Fix

- `test_start = samples_info_sets[test_indices[0]].0`
- `test_end = max(.1)` over the test indices
- Regression test: a property test asserting that **no training label span intersects any test label span** for random variable-length spans, with `pct_embargo = 0`. It must fail before the fix.
- Check whether the unmerged `59ac6fa` (`split_with_diagnostics`, `count_train_test_overlaps`, see #33) already changes this code, and whether `ml_cross_val_score` / feature-importance paths (#27) inherit the window.

## Relations

Found while reviewing #64 (not changed there — that PR is behavior-preserving). Should land before #42 exposes purged CV to Python and before any runbook (#47, #48) relies on it.
https://github.com/Open-Quant/openquant/pull/66

### PR DESCRIPTION
**Pull request by Sean-Koval**
## Scope

[code / P0] `PurgedKFold::split` in `crates/openquant/src/cross_validation.rs` builds its purge window from `.1` (label END) of the first test sample instead of `.0` (label START), and takes the window end from the last test sample rather than the max end over the fold. Training samples whose labels overlap the leading test labels are therefore not purged (label leakage), and variable-length labels can under-purge the trailing edge. In scope: correct the window, add a regression property test that fails before the fix, check every other caller/copy of the same window logic in the crate (the free `ml_get_train_times`-style helper near line 95, feature-importance and hyperparameter-tuning CV paths) and the docs page for the module. Out of scope: Python bindings (#42), re-porting 59ac6fa (#33), changing embargo semantics.

## Acceptance

- A property test over random variable-length label spans with pct_embargo = 0 asserts that no training label span intersects any test label span; it fails on main and passes after the fix.
- A hand-checkable case (daily bars, 3-day labels) asserts the exact expected train indices for each fold.
- Existing cross_validation, feature_importance and hyperparameter_tuning tests pass; any expectation that changes is listed in the PR with the reason it was wrong before.
- Every other occurrence of the window logic in the crate is fixed or shown correct.

Closes #65

### CODE REVIEW

**Comment by Sean-Koval** [comment]
Review notes

**Worse than filed in #65.** The property test showed that with variable-length labels the old window could invert (`test_start` = end of the first test label can be later than `test_end` = end of the last). Then nothing was purged and the **test rows stayed in the training set** — on `main`, the seeded case fails with `train 1` and `test 0` from the same fold. The original code only ever removed test rows as a side effect of purging.

**Blast radius.** `ml_cross_val_score`, feature importance and hyperparameter tuning take their splits from `PurgedKFold`, so any CV score computed with overlapping labels was optimistic. `backtesting_engine::apply_purge_and_embargo` compares spans pairwise per test index and is **not** affected. `ml_get_train_times` is correct given a correct window.

**Behavior changes**
- Fewer training rows per fold whenever labels overlap fold boundaries (the fix).
- `PurgedKFold::new` returns `Err` for `n_splits < 2` or `> n_samples`; both previously panicked.
- Unchanged: embargo semantics (still applied on both sides of the fold, which is more conservative than AFML's after-only embargo — left alone deliberately).

**Evidence.** Two of the three new tests fail on `main` and pass here. `cargo test --workspace`: 246 passed / 0 failed / 1 ignored; no existing expectation changed. `cargo fmt --check` clean; clippy clean under the flags CI uses on `main`.

**Merge note.** #64 also edits `cross_validation.rs` (type alias, loop rewrites). Whichever merges second needs a small conflict resolution; I'd merge this one first since it is the correctness fix.

**Comment by Sean-Koval** [comment]
CI note: `benchmark-regression` failed once on `synthetic_ticker_pipeline_end_to_end` (+53.2% vs a 50% limit) and passed on re-run with no code change. That bench imports only `risk_metrics`, `sampling` and `fast_ewma`, none of which this PR touches, and the same job passed on the previous push of this branch. Shared-runner timing noise — the gate's flakiness is worth its own look under #41.
